### PR TITLE
CBL-2045 Implement Maintenance's Optimize Options

### PR DESCRIPTION
**Pull request by Sandychuang8**
Adding Optimize and FullOptimize options in MaintenanceType

### CODE REVIEW

File: src/LiteCore/src/LiteCore.Shared/Interop/C4Database_native.cs [hunk 1]
```diff
@@ -117,10 +117,6 @@ public static string c4db_getPath(C4Database* db)
         [return: MarshalAs(UnmanagedType.U1)]
         public static extern bool c4db_endTransaction(C4Database* database, [MarshalAs(UnmanagedType.U1)]bool commit, C4Error* outError);
 
-        [DllImport(Constants.DllName, CallingConvention = CallingConvention.Cdecl)]
-        [return: MarshalAs(UnmanagedType.U1)]
-        public static extern bool c4db_isInTransaction(C4Database* database);
```
Comment (borrrden) [inline src/LiteCore/src/LiteCore.Shared/Interop/C4Database_native.cs:122]: What does this have to do with the PR?

Comment (Sandychuang8) [reply src/LiteCore/src/LiteCore.Shared/Interop/C4Database_native.cs:122]: Somehow this PR carries over the commit that I made but not yet pushed to the master branch when I created this PR branch from master branch... The reason I didn't push it is because I don't want a new build just for those changes... Let me take them out and create another PR for it.

**Review by borrrden — APPROVED** [review]
